spec/lib/chat_channel_fetcher_spec.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+describe DiscourseChat::ChatChannelFetcher do
+  describe ".unread_counts" do
+    fab!(:user_1) { Fabricate(:user) }
+    fab!(:user_2) { Fabricate(:user) }
+    fab!(:chat_channel) { Fabricate(:chat_channel) }
+
+    context "user is member of the channel" do
+      before do
+        Fabricate(:user_chat_channel_membership, chat_channel: chat_channel, user: user_1)
+      end
+
+      context "has unread messages" do
+        before do
+          Fabricate(:chat_message, chat_channel: chat_channel, message: "hi", user: user_2)
+          Fabricate(:chat_message, chat_channel: chat_channel, message: "bonjour", user: user_2)
+        end
+
+        it "returns the correct count" do
+          unread_counts = subject.unread_counts([chat_channel], user_1)
+          expect(unread_counts[chat_channel.id]).to eq(2)
+        end
+      end
+
+      context "has no unread messages" do
+        it "returns the correct count" do
+          unread_counts = subject.unread_counts([chat_channel], user_1)
+          expect(unread_counts[chat_channel.id]).to be_nil
```

Comment:
Hmm the return value here feels odd IMO because `nil` isn't a count. Personally, I would prefer we return `0` instead of `nil`.

Reply:
Yes I do agree it came as a surprise

Reply:
How do you feel about this change ?

[unreads_count should always return an integer](https://github.com/discourse/discourse-chat/pull/794/commits/b21233702c2d96ef9e533969715effe505ec6145)